pandas/tests/arrays/string_/test_string.py

```diff
@@ -272,34 +406,49 @@ def test_reduce_missing(skipna):
 
 
 @td.skip_if_no("pyarrow", min_version="0.15.0")
-def test_arrow_array():
+def test_arrow_array(dtype, request):
     # protocol added in 0.15.0
     import pyarrow as pa
 
-    data = pd.array(["a", "b", "c"], dtype="string")
+    if dtype == "arrow_string":
+        reason = (
+            "TypeError: Argument 'other' has incorrect type "
+            "(expected pyarrow.lib.ChunkedArray, got pyarrow.lib.StringArray)"
+        )
+        mark = pytest.mark.xfail(reason=reason)
+        request.node.add_marker(mark)
+
+    data = pd.array(["a", "b", "c"], dtype=dtype)
     arr = pa.array(data)
     expected = pa.array(list(data), type=pa.string(), from_pandas=True)
```

Comment:
Also here I think it's actually working, but the expected needs to be updated. In case of the arrow dtype, this expected needs to be wrapped in a `pa.chunked_array([...])`

Reply:
you are happy with `__arrow_array__` returning the chunked array? The reason for parameterising the current string tests was to keep the behaviour of ArrowStringArray the same as the current StringArray and so the implementation details are hidden from the user.

from https://github.com/pandas-dev/pandas/pull/35259#issuecomment-712399778

> am I correct in assuming that ArrowStringArray and PythonStringArray should behave identically?

Reply:
> you are happy with `__arrow_array__` returning the chunked array?

Yes, we explicitly changed that in pyarrow to accept both arrays and chunked arrays for the `__arrow_array__` protocol (triggered by fletcher using chunked arrays)

>  The reason for parameterising the current string tests was to keep the behaviour of ArrowStringArray the same as the current StringArray and so the implementation details are hidden from the user.

Yes, but conversion to pyarrow is not directly exposed in pandas, so I think that is OK to differ between both StringArray implementations.

Reply:
will update test